v2/pkg/delete/delete_images.go

```diff
@@ -0,0 +1,331 @@
+		o.Log.Error("%v ", err)
+	}
+	var items_map = make(map[string]v1alpha3.DeleteItem)
+
+	// gather related blobs
+	for _, img := range images {
+		copyIS, err := buildFormatedCopyImageSchema(img.Origin, img.Destination, o.LocalStorageFQDN)
+		if err != nil {
+			o.Log.Error("%v ", err)
+		}
+		// clean up the destination url
+		// for our output yaml
+		name := strings.Split(copyIS.Destination, o.LocalStorageFQDN)
+		if len(name) > 0 {
+			copyIS.Destination = name[1][1:]
+		}
+		item := v1alpha3.DeleteItem{
+			ImageName:      copyIS.Origin,
+			ImageReference: copyIS.Destination,
+		}
+		if err != nil {
+			o.Log.Error("%v ", err)
+		}
+		i, err := o.Blobs.GatherBlobs(context.Background(), img.Destination)
+		if err != nil {
+			o.Log.Error("%v image : %s", err, i)
+		}
+		// physically delete blobs
+		if err != nil {
+			o.Log.Error(deleteImagesErrMsg, err)
+		}
+		var blobs []string
+		for k := range i {
+			// get related blobs and remove duplicates
+			blobs = append(blobs, k)
+			if err != nil {
+				o.Log.Error("unable to write blob %s %v", k, err)
+			}
+		}
+		sort.SliceStable(blobs, func(i, j int) bool {
+			return blobs[i] < blobs[j]
+		})
+		item.RelatedBlobs = blobs
+		items_map[copyIS.Destination] = item
+	}
+
+	var items []v1alpha3.DeleteItem
+	// convert back
+	for _, v := range items_map {
+		items = append(items, v)
+	}
+	// sort thi items
+	sort.SliceStable(items, func(i, j int) bool {
+		return items[i].ImageReference < items[j].ImageReference
+	})
+	// marshal to yaml and write to file
+	deleteImageList := v1alpha3.DeleteImageList{
+		Kind:       "DeleteImageList",
+		APIVersion: "mirror.openshift.io/v1alpha2",
+		Items:      items,
+	}
+	ymlData, err := yaml.Marshal(deleteImageList)
+	if err != nil {
+		o.Log.Error(deleteImagesErrMsg, err)
+	}
+	err = os.WriteFile(filename, ymlData, 0755)
+	if err != nil {
+		o.Log.Error(deleteImagesErrMsg, err)
+	}
+	// finally copy the deleteimagesetconfig for reference
+	disc := v1alpha2.DeleteImageSetConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DeleteImageSetConfiguration",
+			APIVersion: "mirror.openshift.io/v1alpha2",
+		},
+		DeleteImageSetConfigurationSpec: v1alpha2.DeleteImageSetConfigurationSpec{
+			Delete: v1alpha2.Delete{
+				Platform:         o.Config.Mirror.Platform,
+				Operators:        o.Config.Mirror.Operators,
+				AdditionalImages: o.Config.Mirror.AdditionalImages,
+			},
+		},
+	}
+	discYamlData, err := yaml.Marshal(disc)
+	if err != nil {
+		o.Log.Error("%v ", err)
+	}
+	err = os.WriteFile(discYamlFile, discYamlData, 0755)
+	if err != nil {
+		o.Log.Error(deleteImagesErrMsg, err)
+	}
+	return nil
+}
+
+// DeleteCacheBlobs - does what it says ;)
+func (o DeleteImages) DeleteCacheBlobs(images v1alpha3.DeleteImageList) error {
+	o.Log.Info("deleting images from local cache")
+	blobPath := filepath.Join(o.LocalStorageDisk, blobsDir, "/")
+	if !o.Opts.Global.DeleteGenerate && o.Opts.Global.ForceCacheDelete {
+		for _, img := range images.Items {
```

Comment:
I'm not sure deleting only the blobs cuts it here. 
Don't we need to do something similar to the deletion of the manifests? ie delete the relevant tags, revisions, etc from .oc-mirror/.cache/docker/registry/v2/repositories/$IMG_REF/_manifests?

Reply:
Actually, to be sure this comment is not misunderstood: I really think it is best to use the containers/image API to perform this task, rather than unknowingly delete files and make the registry corrupt

Reply:
I checked the tag blobs are included in the blobs so as an example look here 

```
imageReference: albo/aws-load-balancer-controller-rhel8@sha256:537fc6c0219e98795bc9560869487bc1605a2556f2a06a9702d81c7fbea35132
  relatedBlobs:
  - sha256:095255f2a987393646f9e6bcb2672719ec3b2a0c831b5f176baa8d622daa4bb7
  - sha256:3a8a017b351a84f1f6699886a2de38082e591bbb3193593303458daccf23a2a9
  - sha256:488dddee86b517ffdaab04d2c624e6c0adf839da9b944197d40e81635b0d7aef
  - sha256:537fc6c0219e98795bc9560869487bc1605a2556f2a06a9702d81c7fbea35132
  - sha256:dc35b837139a95d1b9f7f7b0435a024a74ab972416bdc248f3f608c9f917a753
```
for albo the tag is 537fc6c0219e98795bc9560869487bc1605a2556f2a06a9702d81c7fbea35132
and the related blob is 537fc6c0219e98795bc9560869487bc1605a2556f2a06a9702d81c7fbea35132 - so its going to get deleted

Reply:
Yep that is right. I'm less concerned about space, and more concerned about integrity of the cache registry: if the .oc-mirror/.cache/docker/registry/v2/repositories/$IMG_REF/_manifests contains this digest , while it's been deleted from the blobs, would the cache react in a funny way when we copy this image again after the delete (new M2D+D2M)? or would maybe containers/image remediate automatically by redownloading the blob?